lib/util/makeRequest.ts

```diff
@@ -56,6 +57,18 @@ export type Options = request.CoreOptions &
     readonly errorOnNon200Response: unknown
   }
 
+const promisifiedRequest = (options: Options) =>
+  new Promise<{ err: any; response: request.Response; responseBody: any }>(
+    (resolve, reject) =>
+      request(options, (err, response, responseBody) => {
```

Comment:
also this needs to be async

Reply:
it doesn't havvve to be since it returns a promise, you can await on anything not just async. But I'm going to switch to node-fetch as you suggest